WalletCore/WalletAccessor.swift

```
@@ -152,23 +159,27 @@ public struct WalletAccessor {
         transaction: EthereumTransaction,
         completion: @escaping (BigUInt, BigUInt?) -> Void
     ) {
+        guard let web3 =  web3(on: chain) else { return }
+
         DispatchQueue.global(qos: .userInitiated).async {
             let gasPrice: BigUInt =
-                (try? web3(on: chain).eth.getGasPrice()) ?? transaction.gasPrice
+                (try? web3.eth.getGasPrice()) ?? transaction.gasPrice
             var tx = transaction
             tx.gasPrice = gasPrice
             var opts = options
             opts.gasPrice = .manual(gasPrice)
 
-            web3(on: chain).eth.estimateGasPromise(tx, transactionOptions: opts)
+            web3.eth.estimateGasPromise(tx, transactionOptions: opts)
                 .done(on: DispatchQueue.main) { estimate in
                     completion(gasPrice, estimate)
                 }.cauterize()
         }
     }
 
     public func send(on chain: EthNode, transactionData: TransactionData) throws -> String {
-        let contract = web3(on: chain).contract(
+        guard let web3 =  web3(on: chain) else { return "" }
```

Hmm, would it make sense to throw an error instead? Silently dropping the `send(..)` call on the floor might lead to other bugs / flakiness down the line? Maybe we will have a harder time understanding why `send(..)` sometimes does nothing if there is some future race condition with `web3` being good to go? Dunno...

---

Hmm, so the context is, we should never be null here, and a web3 should be assigned by the time we are here. Maybe I should log the error instead? I dont think there is anything useful to show the user at this point though.